### ISSUE TITLE
Too many goroutines in commit phase affect performance

### ISSUE THREAD

**Issue by youjiali1995** [tikv/client-go#342]
@BusyJay found too many goroutines in commit phase affect performance a lot. The `committer-concurrency` is 1024. The region size on left side is smaller than right side.
![image](https://user-images.githubusercontent.com/14819777/138064568-2e8457a9-3931-47d3-901a-0414d8937cbb.png)
It's not an apple to apple comparison, because larger region results in larger requests and batch, but the throughput of TiKV is the same.
![image](https://user-images.githubusercontent.com/14819777/138065274-e1e1ceca-3db8-4a72-aa21-5de1191be666.png)

Disregarding TiKV, too many goroutines affects GC, schedule of TSO goroutine and so on. We can improve it.
![image](https://user-images.githubusercontent.com/14819777/138066216-3f77d825-729d-463a-bc79-ca23dfde9c19.png)
![image](https://user-images.githubusercontent.com/14819777/138066419-1ded0134-42c3-4bba-b57e-9eb9a21ed15f.png)

**Comment by tiancaiamao** [comment]
`committer-concurrency` is configurable and its default value is not 1024

**Comment by BusyJay** [comment]
> `committer-concurrency` is configurable and its default value is not 1024

I don't think this configuration matters. It's possible to increase the transaction count while decrease this configuration to achieve similar regression.
